src/core/common/new.hpp

```diff
@@ -39,6 +39,6 @@
 
 inline void *operator new(size_t, void *p) throw() { return p; }
 
-void OT_CDECL operator delete(void *, size_t) throw();
+void OT_CDECL operator delete(void *) throw();
```

Comment:
@nibanks, I updated this PR to include both.  Works for me on mac and linux.  Let me know if it works for you on windows.